source/guides/maintenance/custom-app-container.rst

```diff
@@ -0,0 +1,402 @@
+example, the clearlinux/cgit container is built off clearlinux/http:latest
+(that is, it contains a :command:`FROM clearlinux/httpd:latest` line in its Dockerfile),
+which in turn is built off the clearlinux/os-core container. This means all
+three layers need to be locked down against the desired OS version.
+
+Similar to the previous examples, we can lock down both of these containers to
+OS version 30550 by making the following changes to the Dockerfile:
+
+.. code-block:: bash
+
+	diff --git a/httpd/Dockerfile b/httpd/Dockerfile
+	index 6b2a6bf..9df89e4 100644
+	--- a/httpd/Dockerfile
+	+++ b/httpd/Dockerfile
+	@@ -7,7 +7,7 @@ RUN swupd update --no-boot-update $swupd_args
+
+	 # Grab os-release info from the minimal base image so
+	 # that the new content matches the exact OS version
+	-COPY --from=clearlinux/os-core:latest /usr/lib/os-release /
+	+COPY --from=clearlinux/os-core:30550 /usr/lib/os-release /
+
+	 # Install additional content in a target directory
+	 # using the os version from the minimal base
+	@@ -26,7 +26,7 @@ COPY --from=clearlinux/os-core:latest / /os_core_install/
+	 RUN cd / && \
+	     find os_core_install | sed -e 's/os_core_install/install_root/' | xargs rm -d &> /dev/null || true
+
+	-FROM clearlinux/os-core:latest
+	+FROM clearlinux/os-core:30550
+	 MAINTAINER qi.zheng@intel.com
+
+	 COPY --from=builder /install_root /
+
+
+	$ docker build --no-cache=true --build-arg http_proxy=$http_proxy --build-arg https_proxy=$https_proxy -t clearlinux/httpd:30550 httpd/
+
+
+	diff --git a/cgit/Dockerfile b/cgit/Dockerfile
+	index 9a3796d..59260fe 100644
+	--- a/cgit/Dockerfile
+	+++ b/cgit/Dockerfile
+	@@ -7,7 +7,7 @@ RUN swupd update --no-boot-update $swupd_args
+
+	 # Grab os-release info from the minimal base image so
+	 # that the new content matches the exact OS version
+	-COPY --from=clearlinux/httpd:latest /usr/lib/os-release /
+	+COPY --from=clearlinux/httpd:30550 /usr/lib/os-release /
+
+	 # Install additional content in a target directory
+	 # using the os version from the minimal base
+	@@ -22,11 +22,11 @@ RUN source /os-release && \
+	 # file exists on different layers. To minimize docker
+	 # image size, remove the overlapped files before copy.
+	 RUN mkdir /os_core_install
+	-COPY --from=clearlinux/httpd:latest / /os_core_install/
+	+COPY --from=clearlinux/httpd:30550 / /os_core_install/
+	 RUN cd / && \
+	     find os_core_install | sed -e 's/os_core_install/install_root/' | xargs rm -d &> /dev/null || true
+
+	-FROM clearlinux/httpd:latest
+	+FROM clearlinux/httpd:30550
+
+	 COPY --from=builder /install_root /
+	 COPY cgitrc /etc/cgitrc
+
+
+	$ docker build --no-cache=true --build-arg http_proxy=$http_proxy --build-arg https_proxy=$https_proxy -t clearlinux/cgit:30550 cgit/
+
+
+Once the target container is built, the installed OS version can be verified
+by noting the :command:`VERSION_ID` value in the :file:`/usr/lib/os-release`
+file in the container filesystem.
+
+.. code-block:: bash
+

```

Comment:
$ is not needed in the codeblock. Use emphasize-lines syntax to highlight VERSION_ID line.